vpr/src/pack/cluster.cpp

```diff
@@ -712,25 +785,80 @@ static bool is_atom_blk_in_pb(const AtomBlockId blk_id, const t_pb* pb) {
     return false;
 }
 
+static void remove_molecule_from_pb_stats_candidates(t_pack_molecule* molecule,
+                                                     t_pb* pb) {
+    int i;
+    bool found_molecule = false;
+
+    for (i = 0; i < pb->pb_stats->num_feasible_blocks; i++) {
+        if (pb->pb_stats->feasible_blocks[i] == molecule) {
+            found_molecule = true;
+        }
+    }
+
+    if (found_molecule == false) {
+        return;
+    }
+
+    //If the highest gain molecule is the molecule to be removed, just decrement
+    if (pb->pb_stats->feasible_blocks[pb->pb_stats->num_feasible_blocks - 1] == molecule) {
+        pb->pb_stats->num_feasible_blocks--;
+
+        return;
+    }
+
+    //Otherwise, shift the molecules while removing the specified molecule
+    for (int j = 0; j < pb->pb_stats->num_feasible_blocks - 1; j++) {
```

Comment:
logic looks broken, have one loop to find the index, and one loop to shift everything after the index